File: trunk/Som/Learning/ExponentionalFactorFunction.cs
using System;

namespace Som.Learning
{
    public class ExponentionalFactorFunction : LearningFactorFunctionBase
    {
        public double StartLearningRate { get; private set; }
        public double MaxIterationsCount { get; private set; }

        public ExponentionalFactorFunction(double startRate, int maxIterationsCount)
            : base(new double[]{startRate, maxIterationsCount})
        {
            StartLearningRate = startRate;
            MaxIterationsCount = maxIterationsCount;
        }

        public override double GetLearningRate(double iteration)
        {
            return StartLearningRate * Math.Exp(- iteration / MaxIterationsCount);
        }
    }
}

File: JavaSom/Learning/ExponentionalFactorFunction.java
package Learning;

	public class ExponentionalFactorFunction extends LearningFactorFunctionBase
	{
		private double privateStartLearningRate;
		public final double getStartLearningRate()
		{
			return privateStartLearningRate;
		}
		private void setStartLearningRate(double value)
		{
			privateStartLearningRate = value;
		}
		private double privateMaxIterationsCount;
		public final double getMaxIterationsCount()
		{
			return privateMaxIterationsCount;
		}
		private void setMaxIterationsCount(double value)
		{
			privateMaxIterationsCount = value;
		}

		public ExponentionalFactorFunction(double startRate, int maxIterationsCount)
		{
			super(new double[]{startRate, maxIterationsCount});
			setStartLearningRate(startRate);
			setMaxIterationsCount(maxIterationsCount);
		}

		@Override
		public double GetLearningRate(double iteration)
		{
			return getStartLearningRate() * Math.exp(- iteration / getMaxIterationsCount());
		}
	}